trieste/models/gpflow/models.py

```diff
@@ -670,29 +705,82 @@ def update(self, dataset: Dataset) -> None:
         num_data = dataset.query_points.shape[0]
         self.model.num_data = num_data
 
-    def get_inducing_variables(self) -> Tuple[TensorType, TensorType, TensorType, bool]:
+        if self._inducing_point_selector is not None:
+            new_inducing_points = self._inducing_point_selector.update(
+                current_inducing_points, self, dataset
+            )
+            if not tf.reduce_all(
+                tf.math.equal(
+                    new_inducing_points,
+                    current_inducing_points,
+                )
+            ):  # only bother updating if points actually change
+                self._update_inducing_variables(new_inducing_points)
+
+    def _update_inducing_variables(self, new_inducing_points: TensorType) -> None:
+        """
+        When updating the inducing points of a model, we must also update the other
+        inducing variables, i.e. `q_mu` and `q_sqrt` accordingly. The exact form of this update
+        depends if we are using whitened representations of the inducing variables.
+        See :meth:`_whiten_points` for details.
+
+        :param new_inducing_points: The desired values for the new inducing points.
+        :raise NotImplementedError: If we try to update the inducing variables of a model
+            that has more than one set of inducing points.
+        """
+
+        if isinstance(new_inducing_points, list):
+            raise NotImplementedError(
+                f"""
+                We do not currently support updating models with multiple sets of
+                inducing points however received; {new_inducing_points}
+                """
+            )
+
+        old_inducing_points, _, _, whiten = self.get_inducing_variables()
+        tf.assert_equal(
+            tf.shape(old_inducing_points), tf.shape(new_inducing_points)
+        )  # number of inducing points must not change
+
+        if whiten:
+            new_q_mu, new_q_sqrt = _whiten_points(self, new_inducing_points)
+        else:
+            new_q_mu, new_f_cov = self.predict_joint(new_inducing_points)  # [N, L], [L, N, N]
+            jitter_mat = DEFAULTS.JITTER * tf.eye(
+                tf.shape(new_inducing_points)[0], dtype=new_f_cov.dtype
+            )
+            new_q_sqrt = tf.linalg.cholesky(new_f_cov + jitter_mat)
+
+        self.model.q_mu.assign(new_q_mu)  # [N, L]
+        self.model.q_sqrt.assign(new_q_sqrt)  # [L, N, N]
+
+        if isinstance(self.model.inducing_variable, SharedIndependentInducingVariables):
+            self.model.inducing_variable.inducing_variable.Z.assign(new_inducing_points)  # [M, D]
```

Comment:
Is the bulk of this logic specific to `SparseVariational` or is it the same in all `SupportsGetInducingVariables` models? If so, should we perhaps update `SupportsGetInducingVariables` with an abstract `set_inducing_variables(new_inducing_points, new_q_mu, new_q_sqrt)` method, and a concrete `update_inducing_variables(new_inducing_points)` method which calculates the new q values and calls it?

Reply:
these are good questions, quite relevant as we need to move SGPR to its own wrapper...

Reply:
This PR has nothing to do with SGPR. I only support Sparse Variational

Reply:
I dont know how the SGPR works. In fact there are lots of issues with using it at the moment. Namely,  its trajectory_sampler, conditonal_predict* methods, and covariance_between_points all are assuming that it is a GPR. I think we should get rid of it. 

In summary, we never use it and none of the stuff thats meant to work with it does

Reply:
I wont make these changes in this PR